### PR TITLE
CC mods + Native tests + Fix tag/domain serialization

### PR DESCRIPTION
**Pull request by ryobg**
As consequence of work on #16:
- Experimental support for CC mods
- Add native test executor (JCDebug project) - easies the debugging
- Fix, I think, the broken istring serialization, looks used for tags & domain names
- Switched to C++17 features 😍

### CODE REVIEW

File: src/JContainers/src/forms/form_handling.h [hunk 1]
```diff
@@ -9,134 +12,191 @@
...
+}
 
-    inline FormHandle form_id_to_handle(FormId id) {
-        return (FormHandle)(0x0000ffff00000000 | (uint64_t)id);
-    }
+/// Reports the mod index of absolute form id
+inline std::uint8_t mod_index (FormId n) 
+{
+    auto u32 = static_cast<std::uint32_t> (n);
+    return static_cast<std::uint8_t> (is_light (n) ? u32 >> 24 : u32 >> 12);
```
Comment (tasairis) [inline src/JContainers/src/forms/form_handling.h]: Ternary looks backwards.

Comment (ryobg) [reply src/JContainers/src/forms/form_handling.h]: ... that's not even used. Deleted.

File: src/JContainers/src/skse/skse.cpp [hunk 2]
```diff
@@ -138,11 +138,19 @@ struct real_api : public skse_api
...
+    /// the light weight mod index.
+    std::optional<std::uint16_t> loaded_light_mod_index (std::string_view const& name) override
     {
         using namespace std;
-        auto ndx = DataHandler::GetSingleton ()->GetLoadedLightModIndex (string (name).c_str ());
-        return ndx != 0xFF ? make_optional (ndx) : nullopt;
+        auto modinfo = DataHandler::GetSingleton ()->LookupLoadedLightModByName (string (name).c_str ());
+        if (modinfo)
+        {
+            auto ndx = reinterpret_cast<uint16_t const*> (&modinfo->modIndex);
```
Comment (tasairis) [inline src/JContainers/src/skse/skse.cpp]: &modIndex+2

File: src/JContainers/src/skse/skse.cpp [hunk 3]
```diff
@@ -138,11 +138,19 @@ struct real_api : public skse_api
...
+            auto light_ndx = ndx + 1;
```
Comment (tasairis) [inline src/JContainers/src/skse/skse.cpp]: +1? The first ESL is supposed to be 000. That number at modIndex+2 is correct as-is.

**Comment by tasairis** [comment]
Ahhh, nevermind, misread that.

**Comment by ryobg** [comment]
Yeah, maybe not the straight for reading... Well, if you have tried it actuall with your tests then I can assume this page closed?

**Comment by tasairis** [comment]
Yup. Pulled down, rebuilt, and tested the branch and everything works 💯

**Comment by ryobg** [comment]
@tasairis thank you! I do not believe I would have the time to debug all. All kudos to you for this.

**Pull request merged**